### PR TITLE
Remove obsolete classes

### PR DESCRIPTION
**Pull request by stefannikolei**
Removes all obsolete classes as mentioned in #45355

### CODE REVIEW

**Comment by AraHaan** [comment]
hmm looks like this was forgotten about in .NET 7?

File: src/Servers/IIS/IIS/src/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -18,35 +18,8 @@ Microsoft.AspNetCore.Http.Features.IServerVariablesFeature.this[string! variable
 Microsoft.AspNetCore.Http.Features.IServerVariablesFeature.this[string! variableName].set -> void (forwarded, contained in Microsoft.AspNetCore.Http.Features)
 Microsoft.AspNetCore.Server.IIS.BadHttpRequestException
 Microsoft.AspNetCore.Server.IIS.BadHttpRequestException.StatusCode.get -> int
-Microsoft.AspNetCore.Server.IIS.Core.IISServerAuthenticationHandler
```
Comment (BrennanConroy) [inline src/Servers/IIS/IIS/src/PublicAPI.Shipped.txt:21]: Don't touch this file, instead put a
`*REMOVED*Microsoft.AspNetCore.Server.IIS.Core.IISServerAuthenticationHandler` entry in PublicAPI.Unshipped.txt for each entry here.

**Comment by halter73** [comment]
This looks good to me pending API review on #45355. Our normal meeting is tomorrow, and I think we should be able to cover this quickly.

**Comment by adityamandaleeka** [comment]
@halter73 Was this covered in API review today?

**Comment by halter73** [comment]
I had a tab open but forgot to get to it. Fortunately, the next API review is Monday morning, so it shouldn't have to wait too long.

**Comment by AraHaan** [comment]
I wish it was possible to have api review 2 days a week for situations like this where time became a problem for reviews on issues that were planned to be reviewed (or at least extend it) for when there is a massive number of changes that needs reviewed).

**Comment by stefannikolei** [comment]
@halter73 i will have a look at it.

**Comment by stefannikolei** [comment]
@halter73 'ThrowingWasUpgradedWriteOnlyStream' and 'WriteOnlyStream' are already in the MR:

When I look through Servers.IIS only following files are obsolete
- BadHttpRequestException
- HttpContextExtensions.GetIISServerVariable

Should I take care of those too?

**Comment by Tratcher** [comment]
> When I look through Servers.IIS only following files are obsolete
> 
> * BadHttpRequestException
> * HttpContextExtensions.GetIISServerVariable
> 
> Should I take care of those too?

No, people actually use those, we'll wait longer before removing them.

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**